dotCMS/src/main/java/com/dotcms/publisher/endpoint/business/PublishingEndPointFactoryImpl.java

```diff
@@ -135,8 +136,16 @@ public void deleteEndPointById(String id) throws DotDataException {
 	public PublishingEndPoint getEnabledSendingEndPointByAddress(String address) throws DotDataException {
 		ensureCacheIsLoaded();
 		List<PublishingEndPoint> allEndPoints = getEndPoints();
+		String ipOrNetMask;
+		boolean match;
 		for(PublishingEndPoint endPoint : allEndPoints) {
-			if(isMatchingEndpoint(endPoint.getAddress(), address) && endPoint.isEnabled() && endPoint.isSending())
+			ipOrNetMask = endPoint.getAddress();
+			if (ipOrNetMask.contains("/")) {
+				match = new SubnetUtils(ipOrNetMask).getInfo().isInRange(address);
+			} else {
+				match = isMatchingEndpoint(ipOrNetMask, address);
+			}
+			if(match && endPoint.isEnabled() && endPoint.isSending())
```

Comment:
We could extract a bit of this, for example create a "boolean isInRange(address, netmask) method" that checks if the netmask is actually a netmask and in that case do the range check. That way we move some of the responsibility of this method to a new method